src/Rings/MPolyQuo.jl

```diff
@@ -1886,3 +1886,44 @@ end
 
 # extension of common functionality
 symbols(A::MPolyQuoRing) = symbols(base_ring(A))
+
+########################################################################
+# Tensor products of rings
+########################################################################
+
+function tensor_product(A::MPolyRing, B::MPolyRing)
+  kk = coefficient_ring(A)
+  @assert kk === coefficient_ring(B) "coefficient rings do not coincide"
+  all_symb = vcat(symbols(A), symbols(B))
+  r = ngens(A)
+  s = ngens(B)
+  res, _ = polynomial_ring(kk, all_symb, cached=false)
+  return res, hom(A, res, gens(res)[1:r]), hom(B, res, gens(res)[r+1:r+s])
```

Comment:
I think this would be more natural (and allocates less):
```suggestion
  res, a, b = polynomial_ring(kk, symbols(A), symbols(B); cached=false)
  return res, hom(A, res, a), hom(B, res, b)
```

Reply:
Thanks! Didn't know about this constructor.

Reply:
Out of curiosity.  When parents are created in a function we want `cached=false` even if they are part of the return data?  What is the rationale here?